example/weather/services/tester/run_tests.go

```diff
@@ -0,0 +1,155 @@
+package tester
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime/debug"
+	"strings"
+	"sync"
+	"time"
+
+	gentester "goa.design/clue/example/weather/services/tester/gen/tester"
+	"goa.design/clue/log"
+	"golang.org/x/exp/slices"
+)
+
+// Ends a test by calculating duration and appending teh results to the test collection
+func endTest(tr *gentester.TestResult, start time.Time, tc *TestCollection, results []*gentester.TestResult) {
+	elapsed := time.Since(start).Milliseconds()
+	tr.Duration = elapsed
+	results = append(results, tr)
+	tc.AppendTestResult(results...)
+}
+
+// recovers from a panicked test. This is used to ensure that the test
+// suite does not crash if a test panics.
+func recoverFromTestPanic(ctx context.Context, testName string, testCollection *TestCollection) {
+	if r := recover(); r != nil {
+		msg := fmt.Sprintf("[Panic Test]: %v", testName)
+		err := errors.New(msg)
+		log.Errorf(ctx, err, fmt.Sprintf("%v", r))
+		// This doesn't work as I'd like because it only prints to stderr, not to a string
+		// 	that I can caputre and put in the goa log.
+		debug.PrintStack()
+		resultMsg := fmt.Sprintf("%v | %v", msg, r)
+		testCollection.AppendTestResult(&gentester.TestResult{
+			Name:     testName,
+			Passed:   false,
+			Error:    &resultMsg,
+			Duration: -1,
+		})
+	}
+}
+
+// Runs the tests from the testmap and handles filtering/exclusion of tests
```

Comment:
What is the point of the `testCollection` argument if this is just testing the tests in `testMap`?

Reply:
@raphael IIRC (it's been ... awhile since I first patterned this) that *TestCollection param is a pointer and what gets modified as tests in the map are run. It's basically where the results are being built up as each test in the map is run.

Reply:
@raphael IIRC (it's been ... awhile since I first patterned this) that *TestCollection param is a pointer and what gets modified as tests in the map are run. It's basically where the results are being built up as each test in the map is run.

Reply:
@raphael  Nothing in TestCollection is being ignored. 

```
type TestCollection struct {
	// Name of the test collection
	Name string
	// Test results
	Results []*TestResult
	// Duration of the tests in ms
	Duration int64
	// Number of tests that passed
	PassCount int
	// Number of tests that failed
	FailCount int
}
```

All those fields are being used at some point down the line to when it gets returned to the user. It _is_ the dedicated struct for results. Unless I'm missing something in what you're saying (very possible)